### DISCUSSION TITLE
ResourceDictionary in ResourceDictionary

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#11236] [General]
Create a discussion before file an issue as I don't know there to go. 
In current master branch, app will crash upon opening a managed file chooser. The root cause is:
1. A resource dictionary implementing IValueConverter is used in ManagedFileChooser theme. Since it is a static resource in resource dictionary, there is a key assigned to this. See

https://github.com/AvaloniaUI/Avalonia/blob/8ffbbb7dca6b89773da89a54e42bb64bf089d5f5/src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml#L27

https://github.com/AvaloniaUI/Avalonia/blob/8ffbbb7dca6b89773da89a54e42bb64bf089d5f5/src/Avalonia.Dialogs/Internal/ResourceSelectorConverter.cs#L8-L14

2. When compiling xaml, the key is considered as IThemeVariantProvider.Key. See

https://github.com/AvaloniaUI/Avalonia/blob/8ffbbb7dca6b89773da89a54e42bb64bf089d5f5/src/Avalonia.Base/Controls/ResourceDictionary.cs#L124

3. It is automatically converted by this associated converter. 

https://github.com/AvaloniaUI/Avalonia/blob/8ffbbb7dca6b89773da89a54e42bb64bf089d5f5/src/Avalonia.Base/Styling/ThemeVariant.cs#L11-L16

Since `Icons` is an invalid theme name, so it throws exception. 

**Question**

It is very surprising that a ResourceDictionary's key must be ThemeVariant when there is not restriction to define ResourceDictionary as a resource. For this issue we can easily fix it by defining a regular IValueConverter. However, I want to know if it is expected to have such a conflict? Is it better to name `IThemeVariantProvider.Key` as `ThemeKey`?

**Comment by maxkatz6** [comment]
> However, I want to know if it is expected to have such a conflict? 

No, it should try to assign key only if it's in the theme dictionary. So, it's a bug.

> Is it better to name IThemeVariantProvider.Key as ThemeKey?

Current naming seems expected here, as it's specific to IThemeVariantProvider already.
